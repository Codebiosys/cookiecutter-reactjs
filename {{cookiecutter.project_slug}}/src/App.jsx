import React from "react";
import { Route } from "react-router";
import { ConnectedRouter } from "react-router-redux";
import { ApolloProvider } from "react-apollo";
import createHistory from "history/createBrowserHistory";

import apollo from "apolloService";
import storeFactory from "reduxStore";

import AppCanvas from "components/AppCanvas";
import Hello from "containers/Hello";
import Messenger from "containers/Messenger";

const history = createHistory();

const store = storeFactory({ apollo, history });

const App = () => (
  <ApolloProvider client={apollo} store={store}>
    <AppCanvas messenger={<Messenger />}>
      <ConnectedRouter history={history}>
        <div>
          <Route path="/" component={Hello} />
        </div>
      </ConnectedRouter>
    </AppCanvas>
  </ApolloProvider>
);

export default App;
