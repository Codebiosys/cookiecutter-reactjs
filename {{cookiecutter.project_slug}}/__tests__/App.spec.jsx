/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';

import App from 'App';

describe('The App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />,
    );
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
