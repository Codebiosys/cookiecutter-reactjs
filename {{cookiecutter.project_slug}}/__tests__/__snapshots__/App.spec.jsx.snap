// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App renders 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "addTypename": true,
      "dataId": [Function],
      "dataIdFromObject": [Function],
      "disableNetworkFetches": false,
      "fieldWithArgs": [Function],
      "fragmentMatcher": HeuristicFragmentMatcher {},
      "initialState": Object {},
      "middleware": [Function],
      "mutate": [Function],
      "networkInterface": HTTPFetchNetworkInterface {
        "_afterwares": Array [],
        "_middlewares": Array [],
        "_opts": Object {},
        "_uri": "http://:4000/graphql",
      },
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "addTypename": true,
        "deduplicator": Deduplicator {
          "inFlightRequestPromises": Object {},
          "networkInterface": HTTPFetchNetworkInterface {
            "_afterwares": Array [],
            "_middlewares": Array [],
            "_opts": Object {},
            "_uri": "http://:4000/graphql",
          },
        },
        "fetchQueryPromises": Object {},
        "fragmentMatcher": HeuristicFragmentMatcher {},
        "idCounter": 1,
        "lastRequestId": Object {},
        "networkInterface": HTTPFetchNetworkInterface {
          "_afterwares": Array [],
          "_middlewares": Array [],
          "_opts": Object {},
          "_uri": "http://:4000/graphql",
        },
        "observableQueries": Object {},
        "pollingTimers": Object {},
        "queryDeduplication": true,
        "queryDocuments": Object {},
        "queryIdsByName": Object {},
        "queryListeners": Object {},
        "reducerConfig": Object {
          "addTypename": true,
          "customResolvers": undefined,
          "dataIdFromObject": [Function],
          "fragmentMatcher": [Function],
        },
        "reduxRootSelector": [Function],
        "scheduler": QueryScheduler {
          "inFlightQueries": Object {},
          "intervalQueries": Object {},
          "pollingTimers": Object {},
          "queryManager": [Circular],
          "registeredQueries": Object {},
        },
        "ssrMode": false,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
        },
      },
      "reducerConfig": Object {
        "addTypename": true,
        "customResolvers": undefined,
        "dataIdFromObject": [Function],
        "fragmentMatcher": [Function],
      },
      "resetStore": [Function],
      "setStore": [Function],
      "ssrMode": false,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
      },
      "version": "1.8.0",
      "watchQuery": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AppCanvas
    messenger={<Connect(MessengerView) />}
  >
    <ConnectedRouter
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <Route
          component={[Function]}
          path="/"
        />
      </div>
    </ConnectedRouter>
  </AppCanvas>
</ApolloProvider>
`;
